modules/geotiff/src/lib/load-geotiff.ts

```diff
@@ -26,24 +25,26 @@ interface TiffOptions {
  * multi-threaded pool of image decoders should be used to decode tiles (default = true).
  * @return {Promise<{ data: TiffPixelSource[], metadata: ImageMeta }>} data source and associated OME-Zarr metadata.
  */
-export async function loadGeoTiff(source: string | File, opts: TiffOptions = {}) {
+export async function loadGeoTiff(source: string | File | GeoTIFF, opts: TiffOptions = {}) {
```

Comment:
Nits: 
- Since the function is called `fromBlob` can the type be the more general `Blob` instead of `File`?
- I assume we could also accept `ArrayBuffer` and just call `new Blob([arrayBuffer])` (or new `File(...)`)?

Reply:
I think you are correct, and it can be more general `Blob`. There is a `fromArrayBuffer` constructor as well: https://geotiffjs.github.io/geotiff.js/global.html

I'm not sure if you have an opinion on how to include these different global constructors in loaders.

Reply:
I am thinking about it. FWIW:
- loaders.gl is currently standardized on `parse(ArrayBuffer)` and `parseInBatches(AsyncIterator<ArrayBuffer>)`. 
- loaders.gl/core automatically converts from other types (File, Blob, Response, ...) to those types before calling the loaders.

Reply:
That makes sense. However, I think the `fromArrayBuffer` method is for TIFF images that fit into memory (e.g. `fromArrayBuffer(fetch('https://data.ome.tiff').then(res => res.arrayBuffer()))`). The other sources are intended for large, multichannel/tiled TIFFs, where different byte-ranges are read dynamically.

Reply:
Perhaps, similar to Zarr, there would be a way to create a GeoTIFF source using loader.gl utils/modules: https://github.com/geotiffjs/geotiff.js/tree/master/src/source

Reply:
> * loaders.gl is currently standardized on `parse(ArrayBuffer)` and `parseInBatches(AsyncIterator<ArrayBuffer>)`.

What do we do for large tilesets like i3s and 3dtiles? It seems a similar issue here where we don't want to bring the entire tileset into memory.

Reply:
> Note there are even some instances where you want to pass multiple URLs to `fromURLs` to create a single file, for example with a Landsat or Sentinel scene where there are 10 or 11 different files and you might want to load 3 of them.

I don't think it fits very well in the loaders.gl paradigm, but allowing for `loadGeoTIFF` to take _any_ `GeoTIFF` object (regardless of source), enables the flexibility for tiff storage/layout (similar to allowing for a `zarr.Store`).

Reply:
> I don't think it fits very well in the loaders.gl paradigm, but allowing for `loadGeoTIFF` to take _any_ `GeoTIFF` object (regardless of source)

Note this matches my suggestion [here](https://github.com/visgl/loaders.gl/issues/1140#issuecomment-855042176) to avoid unnecessary fetches of the metadata. Essentially we'd have two loaders, a `GeoTIFFMetadataLoader` and a `GeoTIFFLoader`. The `GeoTIFFMetadataLoader` would fetch only the metadata of the GeoTIFF: here calling `fromUrl`, `fromBlob`, etc and returning a `GeoTIFF` object. The `GeoTIFFLoader` would either use the metadata/`GeoTIFF object` passed in or call the `GeoTIFFMetadataLoader` on the input.

This approach makes it possible for advanced users to load multiple tiles in succession while only loading the metadata once. Of course a similar approach is possible for Zarr as well.

Reply:
Definitely something Analogous for Zarr as well. Given the `.zarray` metadata (JSON) and a valid store, you can initialize `ZarrArray` object(s) directly.